rootfs/scheduler/__init__.py

```diff
@@ -993,6 +993,10 @@ def _wait_until_pods_are_ready(self, namespace, container, labels, desired):  #
             logger.info('timed out ({}s) waiting for pods to come up in namespace {}'.format(timeout, namespace))  # noqa
 
         logger.info("{} out of {} pods in namespace {} are in service".format(count, desired, namespace))  # noqa
+        if count != desired:
+            # raising to allow operations to rollback
+            raise KubeException('Not enough pods in namespace {} came into service. '
+                                '{} out of {}'.format(namespace, count, desired))
```

Comment:
Okay, that's fair. Most people will be able to use `deis ps` to do diagnostics but maybe there is a way to get more info into the Exception. Only issue could be overloading the Exception message with a lot of information since you might be deploying a batch of 10 Pods and they all fail.

Before I (personally) didn't see it as a failure but as seen in #706 and #785 that not treating this as a failure is causing problems. The scenario here is this:

I have many releases done via Deis Workflow, I'm at v28 - I do some config setting and v29 kicks off. The Pods for v29 never come up and the timeout kicks in (when waiting for the pods to be ready) - After the timeout the deploy system is made to just go start scaling down v28

At the end of all of that I lost v28 and v29 is all in a Pending state. Now my business is down.

Because of the above I've started considering this a failure state. If your readiness probe is still waiting for external dependencies then you should probably increase `HEALTHCHECK_INITIAL_DELAY` - I have also considered making the 2 minute wait for pod timeout as something people can configure, in addition to the `INITIAL_DELAY`

Does that help explain why I'm opting for failure mode?